src/main/cliapp/src/containers/allelesPage/symbols/SymbolFormEditor.js

```diff
@@ -0,0 +1,110 @@
+import { Button } from "primereact/button";
+import { FormTableWrapper } from "../../../components/FormTableWrapper";
+import { useRef } from "react";
+import { SymbolFormTable } from "./SymbolFormTable";
+
+export const SymbolFormEditor = ({ labelColumnSize, state, dispatch }) => {
+  const tableRef = useRef(null);
+
+  const symbols = [state.allele?.alleleSymbol];
+
+  const createNewSymbolHandler = (e) => {
+    e.preventDefault();
+    const newSymbol = {
+      dataKey: 0,
+      synonymUrl: "",
+      internal: false,
+      nameType: null,
+      formatText: "",
+      displayText: ""
+    }
+
+    dispatch({
+      type: "ADD_OBJECT", 
+      value: newSymbol, 
+      entityType: "alleleSymbol", 
+    })
+  };
+
+  const onRowEditChange = (e) => {
+    return null;
+  };
+
+  const nameTypeOnChangeHandler = (props, event) => {
+    dispatch({ 
+      type: 'EDIT_OBJECT', 
+      entityType: 'alleleSymbol', 
+      field: "nameType", 
+      value: event.target.value
+    });
+  };
+
+  const internalOnChangeHandler = (props, event) => {
+    dispatch({ 
+      type: 'EDIT_OBJECT', 
+      entityType: 'alleleSymbol', 
+      field: "internal", 
+      value: event.target?.value?.name
+    });
+  };
+
+  const synonymScopeOnChangeHandler = (props, event) => {
+    dispatch({ 
```

Comment:
Need the following two lines before the dispatch function:
`//temporary solution -- replace with props.editorCallback() after PrimeReact upgrade 
    props.rowData.synonymScope = event.target.value;`
Also need to do the equivalent for internalOnChangeHandler and nameTypeOnChangeHandler.

Reply:
applied